doc/en/user/source/community/oauth2/index.rst

```diff
@@ -0,0 +1,151 @@
+.. _security_tutorials_oauth2:
+
+Authentication with OAuth2
+==========================
+
+This tutorial introduces GeoServer OAuth2 support and walks through the process of
+setting up authentication aganist an OAuth2 provider. It is recommended that the 
+:ref:`security_auth_chain` section be read before proceeding.
+
+OAuth2 Protocol and GeoServer OAuth2 core module
```

Comment:
Most of this section is too technical, this is user documentation (I don't understand it myself, someone that has been exposed to the details of a auth protocol implementation probably would). It's still good material, but should not be the first thing a user reads. Maybe move it at the very end of the document, just telling users that they need to install the core module plus the specific binding one, and if they don't find the one they want, they can go and reference the new section at the end.